redaxo/src/addons/structure/functions/function_rex_url.php

```diff
@@ -57,10 +57,21 @@ function rex_getUrl($id = null, $clang = null, array $params = [], $separator =
 /**
  * Leitet auf einen anderen Artikel weiter.
  *
+ * @param int       $article_id
+ * @param null|int  $clang
+ * @param array     $params
+ *
+ * @throws InvalidArgumentException
+ *
  * @package redaxo\structure
  */
 function rex_redirect($article_id, $clang = null, array $params = [])
 {
+    // article id must be integer
+    if (!is_int($article_id)) {
+        throw new InvalidArgumentException('$article_id is not integer');
```

Comment:
Unser übliches Format für solche Exceptions sieht so aus: https://github.com/redaxo/redaxo/blob/b61af532daca9f30db985629e440c74225cc5787/redaxo/src/core/lib/fragment.php#L90
Also insbesondere geben wir den tatsächlichen Typen an.

Reply:
Nach dem Hintergrund, dass man den Type ignorieren möchte würde ich folgende msg vorschlagen: `sprintf('Expecting $article_id must contain integer rex_article_id, %s is not an id!', $article_id)`